File: flask/src/controller/assistance_controller.py
from flask import request
from response.response_error import ResponseError
from response.response import response
from service.assistance_service import get_assistance_tools, create_assistance_tools

# Controller untuk menangani permintaan GET untuk alat bantuan
def get_assistance_tools_controller(id):
    try:
        result = get_assistance_tools(id)
        return response(200, result, "Get assistance tools success")
    except ResponseError as e:
        raise e

# Controller untuk menangani permintaan POST untuk membuat alat bantuan
def create_assistance_tools_controller():
    try:
        data = request.get_json()
        result = create_assistance_tools(data)
        return response(200, result, "Create assistance tools success")
    except ResponseError as e:
        raise e

File: express/src/controller/assistance-controller.js
import assistanceService from "../service/assistance-service.js";
import { response } from "../response/response.js";

// Controller untuk mengambil bantuan berdasarkan ID
const getAssistanceById = async (req, res, next) => {
  try {
    const result = await assistanceService.getAssistanceById(req.params.id);
    response(200, result, "Get assistance success", res);
  } catch (error) {
    next(error);
  }
};

// Controller untuk membuat alat bantuan baru
const createAssistanceTools = async (req, res, next) => {
  try {
    const result = await assistanceService.createAssistanceTools(req.body);
    response(201, result, "Create assistance tools success", res);
  } catch (error) {
    next(error);
  }
};

export default { getAssistanceById, createAssistanceTools };
